matchmake-extension/auto_matchmake_with_search_criteria_postpone.go

```diff
@@ -0,0 +1,126 @@
+		// This should in theory be impossible, as there aren't enough PIDs creating sessions to fill the uint32 limit.
+		// If we ever get here, we must be not deleting sessions properly
+		if sessionIndex == 0 {
+			logger.Critical("No gatherings available!")
+			return
+		}
+
+		session := common_globals.CommonMatchmakeSession{
+			SearchCriteria: lstSearchCriteria,
+			GameMatchmakeSession:   matchmakeSession,
+		}
+
+		common_globals.Sessions[sessionIndex] = &session
+		common_globals.Sessions[sessionIndex].GameMatchmakeSession.Gathering.ID = sessionIndex
+		common_globals.Sessions[sessionIndex].GameMatchmakeSession.Gathering.OwnerPID = client.PID()
+		common_globals.Sessions[sessionIndex].GameMatchmakeSession.Gathering.HostPID = client.PID()
+
+		common_globals.Sessions[sessionIndex].GameMatchmakeSession.StartedTime = nex.NewDateTime(0)
+		common_globals.Sessions[sessionIndex].GameMatchmakeSession.StartedTime.UTC()
+		matchmakeSession.SessionKey = []byte("00000000000000000000000000000000")
+	}
+
+	common_globals.Sessions[sessionIndex].ConnectionIDs = append(common_globals.Sessions[sessionIndex].ConnectionIDs, client.ConnectionID())
+	common_globals.Sessions[sessionIndex].GameMatchmakeSession.ParticipationCount = uint32(len(common_globals.Sessions[sessionIndex].ConnectionIDs))
+
+	rmcResponseStream := nex.NewStreamOut(server)
+	matchmakeDataHolder := nex.NewDataHolder()
+	matchmakeDataHolder.SetTypeName("MatchmakeSession")
+	matchmakeDataHolder.SetObjectData(common_globals.Sessions[sessionIndex].GameMatchmakeSession)
+	rmcResponseStream.WriteDataHolder(matchmakeDataHolder)
+
+	rmcResponseBody := rmcResponseStream.Bytes()
+
+	rmcResponse := nex.NewRMCResponse(matchmake_extension.ProtocolID, callID)
+	rmcResponse.SetSuccess(matchmake_extension.MethodAutoMatchmakeWithSearchCriteria_Postpone, rmcResponseBody)
+
+	rmcResponseBytes := rmcResponse.Bytes()
+
+	var responsePacket nex.PacketInterface
+
+	if server.PRUDPVersion() == 0 {
+		responsePacket, _ = nex.NewPacketV0(client, nil)
+		responsePacket.SetVersion(0)
+	} else {
+		responsePacket, _ = nex.NewPacketV1(client, nil)
+		responsePacket.SetVersion(1)
+	}
+	responsePacket.SetSource(0xA1)
+	responsePacket.SetDestination(0xAF)
+	responsePacket.SetType(nex.DataPacket)
+	responsePacket.SetPayload(rmcResponseBytes)
+
+	responsePacket.AddFlag(nex.FlagNeedsAck)
+	responsePacket.AddFlag(nex.FlagReliable)
+
+	server.Send(responsePacket)
+
+	rmcMessage := nex.NewRMCRequest()
+	rmcMessage.SetProtocolID(notifications.ProtocolID)
+	rmcMessage.SetCallID(0xffff0000 + callID)
+	rmcMessage.SetMethodID(notifications.MethodProcessNotificationEvent)
+
+	category := notifications.NotificationCategories.Participation
+	subtype := notifications.NotificationSubTypes.Participation.NewParticipant
+
+	oEvent := notifications_types.NewNotificationEvent()
+	oEvent.PIDSource = common_globals.Sessions[sessionIndex].GameMatchmakeSession.Gathering.HostPID
+	oEvent.Type = notifications.BuildNotificationType(category, subtype)
```

Comment:
This function isn't available in the current go modules of the package. Please update them.

Also please update the rest of notifications to use the new format